app/main.py

```diff
@@ -2,24 +2,82 @@
 
 
 class IntegerRange:
-    pass
+
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+        self.private_name = None
+
+    def __set_name__(self, owner: str, name: str) -> None:
+        self.private_name = f"_{name}"
+
+    def __get__(self, instance: str, owner: None = None) -> int:
+        if instance is None:
+            return self
+
+        return getattr(instance, self.private_name, None)
+
+    def __set__(self, instance: str, value: int) -> None:
+        if not isinstance(value, int):
+            raise TypeError(f"Value must be an integer, not {type(value)}")
+
+        if value < self.min_amount or value > self.max_amount:
+            raise ValueError(
+                f"Value must be between {self.min_amount}"
+                f"and {self.max_amount}"
+            )
+
+        setattr(instance, self.private_name, value)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+
+    age = None
+    weight = None
+    height = None
+
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+
+    age = IntegerRange(14, 60)
+    height = IntegerRange(120, 220)
+    weight = IntegerRange(50, 120)
 
 
 class Slide:
-    pass
+
+    def __init__(
+        self, name: str,
+        limitation_class: SlideLimitationValidator
+    ) -> None:
+        self.name = name
+        self.limitation_class = limitation_class
+
+    def can_access(self, visitor: str) -> bool:
```

Comment:
The `visitor` parameter is currently typed as `str`, which is incorrect. It should be of type `Visitor` to match the attributes being accessed (`age`, `weight`, `height`). Update the type hint to `Visitor`.